src/server/snapshot.cc

```diff
@@ -332,14 +330,16 @@ size_t SliceSnapshot::FlushSerialized(SerializerBase::FlushState flush_state) {
   error_code ec = serializer_->FlushToSink(&sfile, flush_state);
   CHECK(!ec);  // always succeeds
 
+  // Yield after possibly long cpu slice due to compression and serialization
+  if (ThisFiber::Priority() == fb2::FiberPriority::BACKGROUND)
+    ThisFiber::Yield();
+  // TODO: else Sleep() to provide write backpressure in advance?
```

Comment:
Previously we yielded after the write, now we yield before because that's the end of the "CPU" slice (that the scheduler masures). There shoudn't be any difference as long as writes are not async (not sure why we didn't add that yet) 🤔